source/simulator/parameters.cc

```diff
@@ -302,6 +302,42 @@ namespace aspect
                        "the composition system gets solved. See 'linear solver "
                        "tolerance' for more details.");
 
+
+    prm.enter_subsection("Formulation");
+    {
+      prm.declare_entry ("Formulation", "custom",
+                         Patterns::Selection ("isothermal compression|custom|ALA|BA"),
+                         "Select a formulation for the basic equations. Different "
+                         "published formulations are available in ASPECT (see list of "
+                         "possible parameters for available options). Two ASPECT specific "
+                         "options are: 'isothermal compression': ASPECT's original formulation, "
+                         "using the incompressible or explicit compressible mass equation, "
+                         "and the full density for the temperature equation. 'custom': "
+                         "A custom selection of 'Mass density approximation' and "
+                         "'Temperature density approximation'. Warning: The 'custom' option is "
+                         "implemented for advanced users that want full control over the "
+                         "equations solved. It is possible to choose inconsistent formulations "
+                         "and no error checking is performed on the consistency of the resulting "
+                         "equations.");
+
+      prm.declare_entry ("Mass conservation", "ask material model",
+                         Patterns::Selection ("incompressible|isothermal compression|"
+                                              "reference density profile|implicit reference density profile|"
+                                              "ask material model"),
+                         "Possible approximations for the density derivatives in the mass "
+                         "conservation equation. Note that this parameter is only evaluated "
+                         "if 'Formulation' is set to 'custom'. Other formulations overwrite "
+                         "this parameter.");
+      prm.declare_entry ("Temperature equation", "real density",
```

Comment:
I think for all of these parameters, we will eventually have to have a section in the manual that explains how they map to the various approximations in the new section you wrote as part of this PR. It's difficult for me to envision how these approximations are mapped to input parameters.